src/java/org/apache/cassandra/security/FileBasedSslContextFactory.java

```diff
@@ -178,10 +173,29 @@ protected TrustManagerFactory buildTrustManagerFactory() throws SSLException
         }
     }
 
+    protected boolean checkExpiredCerts(KeyStore ks) throws KeyStoreException
+    {
+        boolean hasExpiredCerts = false;
+        for (Enumeration<String> aliases = ks.aliases(); aliases.hasMoreElements(); )
+        {
+            String alias = aliases.nextElement();
+            if (ks.getCertificate(alias).getType().equals("X.509"))
+            {
+                Date expires = ((X509Certificate) ks.getCertificate(alias)).getNotAfter();
+                if (expires.before(new Date()))
```

Comment:
Could not this be instantiated just once? final Date date = new Date(); outside of whole for loop. If there is 500 aliases, we will do 500x "new Date()".

Reply:
Great point Stefan. Again thanks for making the change on the separate branch.